akka-cluster-tools/src/main/scala/akka/cluster/pubsub/DistributedPubSubMediator.scala

```diff
@@ -330,6 +330,8 @@ object DistributedPubSubMediator {
           remove(ref)
           context.parent ! Unsubscribed(UnsubscribeAck(msg), sender())
         case Terminated(ref) ⇒
+          //this doesn't seem to be called anymore since Topic business logic overrides it
+          //causing actors to leave in subscribers map after they have died
```

Comment:
Good catch. This comment is a bit confusing here though, since handing `Terminated` is not overridden in `Group`, so this code will be called for that case.

Reply:
Yeah, we'll probably want to amend this commit, the comment was more to quickly point out where the problem appears to be. Here's a piece of code I've been using to reproduce. After the publish (after the actor is dead) you'll see output about deadletters to the actor. We've also checked the heap on production systems with visualvm and see a lot of dead actors in the subscribers set.

```
public class PubSubDeathWatchTest {

	public static void main(String[] args) {
		
		ActorSystem as = ActorSystem.create();
		String topic = UUID.randomUUID().toString();
		ActorRef mediator = DistributedPubSub.get(as).mediator();
		
		ActorRef subscriber = as.actorOf(Props.create(Subscriber.class));
		System.out.println("subscriber: " + subscriber);
		mediator.tell(new Subscribe(topic, subscriber), ActorRef.noSender());
		
		as.scheduler().scheduleOnce(Duration.create(5000, TimeUnit.MILLISECONDS), () -> {
			as.stop(subscriber);
		}, as.dispatcher());
		
		as.scheduler().scheduleOnce(Duration.create(10000, TimeUnit.MILLISECONDS), () -> {
			mediator.tell(new Publish(topic, "foo"), ActorRef.noSender());
		}, as.dispatcher());
	}

	static class Subscriber extends AbstractActor {
		public Receive createReceive() {
			return receiveBuilder().build();
		}
	}
}
```